astlib/migrate_501_502.ml

```diff
@@ -0,0 +1,1262 @@
+module From = Ast_501
+module To = Ast_502
+
+let rec copy_toplevel_phrase :
+    Ast_501.Parsetree.toplevel_phrase -> Ast_502.Parsetree.toplevel_phrase =
+  function
+  | Ast_501.Parsetree.Ptop_def x0 ->
+      Ast_502.Parsetree.Ptop_def (copy_structure x0)
+  | Ast_501.Parsetree.Ptop_dir x0 ->
+      Ast_502.Parsetree.Ptop_dir (copy_toplevel_directive x0)
+
+and copy_toplevel_directive :
+    Ast_501.Parsetree.toplevel_directive -> Ast_502.Parsetree.toplevel_directive
+    =
+ fun {
+       Ast_501.Parsetree.pdir_name;
+       Ast_501.Parsetree.pdir_arg;
+       Ast_501.Parsetree.pdir_loc;
+     } ->
+  {
+    Ast_502.Parsetree.pdir_name = copy_loc (fun x -> x) pdir_name;
+    Ast_502.Parsetree.pdir_arg = Option.map copy_directive_argument pdir_arg;
+    Ast_502.Parsetree.pdir_loc = copy_location pdir_loc;
+  }
+
+and copy_directive_argument :
+    Ast_501.Parsetree.directive_argument -> Ast_502.Parsetree.directive_argument
+    =
+ fun { Ast_501.Parsetree.pdira_desc; Ast_501.Parsetree.pdira_loc } ->
+  {
+    Ast_502.Parsetree.pdira_desc = copy_directive_argument_desc pdira_desc;
+    Ast_502.Parsetree.pdira_loc = copy_location pdira_loc;
+  }
+
+and copy_directive_argument_desc :
+    Ast_501.Parsetree.directive_argument_desc ->
+    Ast_502.Parsetree.directive_argument_desc = function
+  | Ast_501.Parsetree.Pdir_string x0 -> Ast_502.Parsetree.Pdir_string x0
+  | Ast_501.Parsetree.Pdir_int (x0, x1) ->
+      Ast_502.Parsetree.Pdir_int (x0, Option.map (fun x -> x) x1)
```

Comment:
If that were not automatically generated, I would advise to remove the useless `map` :)